lib/actions/actionsBuilder.js

```diff
@@ -0,0 +1,30 @@
+const AddLinkToProjectAction = require('./AddLinkToProjectAction');
+const assert = require('assert');
+
+class ActionsBuilder {
+  constructor(){
+    this.commandMap = {
+      'add': new AddLinkToProjectAction,
+    }
+  }
+
+  parseActionTerm(userInputString = '') {
+    assert(typeof userInputString === 'string');
+
+    let parsedUserInput = userInputString.toLowerCase().split(' ');
+
+    if (parsedUserInput.length < 3){
+      throw new TypeError('invalid input');
+    }
+
+    let actionName = parsedUserInput[0]; 
+    return actionName;
+  }
+
+  map(actionCommand) {
```

Comment:
I would also recommend this method be `static`.

Reply:
Hey @colbyallenc, replying here so @cromwellryan can see. I believe Ryan means that `parseActionTerm` (not `map` / `fromMessageText`) should be a static method called on the Actions class (e.g. `Actions.parseActionTerm`) because it is a utility method that doesn't need to know about the state of an `Actions` instance.

Reply:
thanks for your feedback @yosevukilonzo 
I was thinking about that... 
What do you think if about what I've done here? --> 
`
class Actions {
  constructor(commandMap){
    this.commandMap = commandMap
  }
                 
  parseActionTerm(userInputString) {
    assert(typeof userInputString === 'string');

    if (userInputString === ''){
      throw new TypeError('No Term Found');
    }

    let parsedUserInput = userInputString.toLowerCase().split(' ');

    let actionName = parsedUserInput[0]; 
    return actionName
  }

  fromMessageText(actionCommand) {
    let actionName = this.parseActionTerm(actionCommand)
    return this.commandMap[actionName]
  }

  static botActions(){
    const commandMap = {
      'add': new AddLinkToProjectAction()
    }
    return new Actions(commandMap)
  }
}
`

Reply:
`fromMessageText` must be static and `botActions` should be an externally used function.